### ISSUE TITLE
Tracking too many large files

### ISSUE THREAD

**Issue by andrewwinters5000** [trixi-framework/HOHQMesh#46]
The repo tracks several mesh and plotting `.tec` files (either in the `Benchmarks` or `Examples` folders) causing it to grow in size over time. Here is an list of the largest files tracked by the repo
```
19M ./Benchmarks/PlotFiles/Benchmarks/SeaMount.tec
14M ./Benchmarks/PlotFiles/Benchmarks/SeaMountCubic.tec
13M ./Examples/3D/Pond/Pond3D.tec
6.5M ./Examples/3D/ScaledCylinder/CylinderScale.mesh
5.4M ./Examples/3D/ScaledCylinder/CylinderScale.tec
5.1M ./Examples/3D/HalfCircleRotated/HalfCircle3DR.mesh
5.0M ./Examples/3D/HalfCircleRotated/HalfCircle3DR.tec
5.0M ./Benchmarks/MeshFiles/Benchmarks/SeaMount.mesh
4.4M ./Examples/3D/CavityRampExtruded/CavityRamp3D.tec
4.0M ./Benchmarks/MeshFiles/Benchmarks/SeaMountCubic.mesh
3.7M ./Examples/3D/Pond/Pond3D.mesh
3.4M ./Examples/3D/HalfCircleExtruded/HalfCircle3D.tec
3.0M ./Examples/3D/MtStHelens/sthelens_grid_data.txt
2.4M ./Examples/2D/LakeSuperior/SuperiorMain.tec
1.8M ./Examples/3D/BoxRotated/Box3DRot.mesh
1.8M ./Examples/3D/Box/Box3D.mesh
1.8M ./Examples/2D/GingerbreadMan/GingerbreadManPlot.tec
1.7M ./Examples/2D/EastCoastUS/EastCoastUS2d.tec
1.5M ./Examples/3D/CavityRampExtruded/CavityRamp3D.mesh
1.5M ./Examples/3D/BoxRotated/Box3D.tec
1.4M ./Examples/3D/BottomFromFile/BottomFromFile.tec
1.2M ./Examples/2D/IndianOcean/IndianOcean.mesh
```

 As more examples are added with future development the size of the repo will become quite large and unwieldy to work with. For instance, with my recent attempt to build a new tarball and release a new version tag, `Utilities/createrelease` created `HOHQMesh-v1.3.0.tar.gz` that is 97.5M. This is too large to attach to a github release.

**Comment by sloede** [comment]
🙀🙀🙀

To be honest, I'd suggest to remove all examples larger than a given size and/or move them to a separate repository. I would also suggest to remove all tec files since they blow up the repo size disproportionately. Then, we should bite the bullet and rewrite repo history using the [BFG repo cleaner](https://rtyley.github.io/bfg-repo-cleaner/), but only since the last release (which is bad enough).

**Comment by gregorgassner** [comment]
Would it make sense to just put these files into a separate git repo? with "examples" etc?

**Comment by andrewwinters5000** [comment]
Most of the 2D meshes and plot files are smallish, say less than 4MB. The 3D plot files can be the real killer. For example, the Mt. St. Helens SEM formatted plot file is approximately 200MB. Altogether I think we can safely assume (for the time being) that it is less than a few GB

**Comment by andrewwinters5000** [comment]
> Great. I'll get something up today.

Sounds good, I have untracked everything in PR #47 but all the files still live on `main` for the time being.

**Comment by sloede** [comment]
Not wanting to crash the party, but why not use another git repo, e.g., `HOHQMesh_examples`? The issue I see with using GCS that it boils down to @fluidnumerics-joe having to handle the file management all by himself (uploading new files, deleting unused files, replacing changed files). Thus I suggest to at least give it one more try to find a solution that distributes the responsibility among a few more shoulders. Maybe @andrewwinters5000 has something like Sciebo, where you can upload up to 500 GiBs of files and share access with others?

Having said that, if it is "just" example files that are not referenced on a per-file basis from the repo (i.e., not used during, e.g., CI testing), it might be OK to use some external storage that is managed by a single person, although I do not recommend it.

**Comment by sloede** [comment]
> I am currently investigating for a workaround.

Maybe something along the lines of
```bash
if [[ "$(git status --ignored --porcelain | grep -v '.DS_Store')" != "!! Contrib/FTObjectLibrary/" ]]; then
```
?

However, maybe you could also add [here](https://github.com/trixi-framework/HOHQMesh/blob/d8cdb7283328cbe52ea1326c3d8cd9cfbf439b96/Utilities/createrelease#L48) something like
```bash
# Delete .DS_Store files (relevant on macOS only)
printf "Delete .DS_Store files... "
find "$releasedir" -name .DS_Store -exec rm {} \;
echo "OK"
```

**Comment by andrewwinters5000** [comment]
This still throws an error where DS_Store is reported, e.g.,
```
!! .DS_Store
!! Benchmarks/.DS_Store
...
```
and the creation fails.

**Comment by sloede** [comment]
No, I mean you need both: The first change will make sure that the check passes, the second one ensures that you do not include .DS_Store files in the release.

**Comment by andrewwinters5000** [comment]
I added both, it is what gave me the error

**Comment by andrewwinters5000** [comment]
Also, if we chose to ignore, e.g., `*.tec` we need similar workarounds such that the `createrelease` command does not spuriously think that the current folder is "unclean"

**Comment by sloede** [comment]
> I added both, it is what gave me the error

What is the output of `git status --ignored --porcelain`?

> Also, if we chose to ignore, e.g., `*.tec` we need similar workarounds such that the `createrelease` command does not spuriously think that the current folder is "unclean"

Isn't it the other way around? Since we do *not* want to include those files in our repo anymore, they also shouldn't be part of a release. The unclean check was originally intended to prevent exactly this - I don't know why I commented it out 🤷‍♂️

**Comment by andrewwinters5000** [comment]
> What is the output of `git status --ignored --porcelain`?

It is always a bit annoying editing / testing the `createrelease`. Since it is tracked making changes to is causes `git status --porcelain` to fire.

I tried adding the following to remove these file types from the tarball
```bash
# Delete mesh and tec plotting files
printf "Delete mesh and tec files... "
find "$releasedir" -name *.mesh -exec rm {} \;
find "$releasedir" -name *.inp -exec rm {} \;
find "$releasedir" -name *.tec -exec rm {} \;
echo "OK"
```
This shrinks it significantly to be around 2Mb

**Comment by sloede** [comment]
The problem with adding those lines is that they delete also files that are currently committed in `main`. IMHO a release should usually contain all files in the repo, as otherwise one could ask why we have them in the repo at all? But this is my personal opinion and I wouldn't want to meddle with your release maintainer mojo 😬

**Comment by andrewwinters5000** [comment]
> IMHO a release should usually contain all files in the repo

I totally agree. But if we were to untrack these files as in #47 wouldn't this be okay? Then a user could have local copies of mesh and tecplot files as they use HOHQMesh but building the release would not include them.

 > What is the output of git status --ignored --porcelain?

I realized I didn't fully answer this. From `main` on my machine this is the output:
```
?? Examples/3D/MtStHelens/MtStHelens.inp
?? Examples/3D/MtStHelens/MtStHelens.tec
!! .DS_Store
!! Benchmarks/.DS_Store
!! Benchmarks/BenchmarkData/.DS_Store
!! Benchmarks/MeshFiles/.DS_Store
!! Benchmarks/MeshFiles/Benchmarks/.DS_Store
!! Benchmarks/MeshFiles/Tests/.DS_Store
!! Benchmarks/PlotFiles/.DS_Store
!! Benchmarks/PlotFiles/Benchmarks/.DS_Store
!! Benchmarks/PlotFiles/Tests/.DS_Store
!! Benchmarks/StatsFiles/.DS_Store
!! Benchmarks/StatsFiles/Tests/.DS_Store
!! Contrib/FTObjectLibrary/
!! Documentation/docs/authors.md
!! Documentation/docs/building-the-documentation.md
!! Documentation/docs/index.md
!! Documentation/docs/license.md
!! Examples/.DS_Store
!! Examples/2D/.DS_Store
!! Examples/3D/.DS_Store
!! Examples/3D/BottomFromFile/.DS_Store
!! Examples/3D/MtStHelens/.DS_Store
!! Examples/3D/Pond/.DS_Store
!! Examples/3D/Snake/.DS_Store
!! Source/.DS_Store
!! Source/Project/.DS_Store
```

**Comment by sloede** [comment]
> But if we were to untrack these files as in #47 wouldn't this be okay?

Yes, if y'all decide that we will have absolutely no `.inp`/`.tec`/`.mesh` files in the repo, this would be ok.

**Comment by sloede** [comment]
FYI, I just remembered we had the issue of large(ish) example files also for [ReadVTK.jl](https://github.com/trixi-framework/ReadVTK.jl), where we used a separate repo [ReadVTK_examples](https://github.com/trixi-framework/ReadVTK_examples) to hold the example files.
